src/datasets/load.py

```diff
@@ -1574,46 +1584,64 @@ def dataset_module_factory(
     # Try remotely
     elif is_relative_path(path) and path.count("/") <= 1:
         try:
-            _raise_if_offline_mode_is_enabled()
-            hf_api = HfApi(config.HF_ENDPOINT)
+            # Get the Dataset Card + get the revision + check authentication all at in one call
+            # We fix the commit_hash in case there are new commits in the meantime
+            api = HfApi(
+                endpoint=config.HF_ENDPOINT,
+                token=download_config.token,
+                library_name="datasets",
+                library_version=__version__,
+                user_agent=get_datasets_user_agent(download_config.user_agent),
+            )
             try:
-                dataset_info = hf_api.dataset_info(
+                _raise_if_offline_mode_is_enabled()
+                dataset_readme_path = api.hf_hub_download(
                     repo_id=path,
+                    filename=config.REPOCARD_FILENAME,
+                    repo_type="dataset",
                     revision=revision,
-                    token=download_config.token,
-                    timeout=100.0,
+                    proxies=download_config.proxies,
                 )
+                commit_hash = os.path.basename(os.path.dirname(dataset_readme_path))
+            except EntryNotFoundError as e:
+                if "internet connection" in str(e).lower():
+                    raise ConnectionError(f"Couldn't reach '{path}' on the Hub ({e.__class__.__name__})") from e
+                commit_hash = api.dataset_info(
+                    path,
+                    revision=revision,
+                    timeout=100.0,
+                ).sha
```

Comment:
```py
            except EntryNotFoundError as e:
                if "internet connection" in str(e).lower():
                    raise ConnectionError(f"Couldn't reach '{path}' on the Hub ({e.__class__.__name__})") from e
```

Do you have an example of `EntryNotFoundError` where the body contains `"internet connection"` and the initial error is not explicit enough? I'm a bit wary of relying on exception message as it can be changed without prior notice in `huggingface_hub`. Also if the problem is that the exception message is not good enough, it's preferable to fix it in `huggingface_hub` directly.

Reply:
this once is when the connection to HF fails, it raises LocalEntryNotFoundError because it can't find the cached file when there is no connection

Reply:
the error comes from here: https://github.com/huggingface/huggingface_hub/blob/e118ecc8544533a3a43f43ff3b5dca43274d1dd0/src/huggingface_hub/file_download.py#L1486-L1491

Reply:
And isn't the `raise LocalEntryNotFoundError from ...` enough for the users?